pkg/arvo/app/graph-push-hook.hoon

```diff
@@ -0,0 +1,116 @@
+/+  store=graph-store
+/+  res=resource
+/+  graph
+/+  group
+/+  default-agent
+/+  dbug
+/+  push-hook
+~%  %graph-push-hook-top  ..is  ~
+|%
++$  card  card:agent:gall
+++  config
+  ^-  config:push-hook
+  :*  %graph-store
+      /updates
+      update:store
+      %graph-update
+      %graph-pull-hook
+  ==
+::
++$  agent  (push-hook:push-hook config)
+--
+::
+%-  agent:dbug
+^-  agent:gall
+%-  (agent:push-hook config)
+^-  agent
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+    grp   ~(. group bowl)
+    gra   ~(. graph bowl)
+::
+++  on-init   on-init:def
+++  on-save   !>(~)
+++  on-load   on-load:def
+++  on-poke   on-poke:def
+++  on-agent  on-agent:def
+++  on-watch  on-watch:def
+++  on-leave  on-leave:def
+++  on-peek   on-peek:def
+++  on-arvo   on-arvo:def
+++  on-fail   on-fail:def
+::
+++  should-proxy-update
+  |=  =vase
+  ^-  ?
+  =/  =update:store  !<(update:store vase)
+  ?-  -.q.update
+      %add-graph          %.y
+      %remove-graph       %.y
+      %add-nodes          %.y
+      %remove-nodes       %.y
+      %add-signatures     %.y
+      %remove-signatures  %.y
+      %archive-graph      %.y
+      %unarchive-graph    %.n
+      %add-tag            %.n
+      %remove-tag         %.n
+      %keys               %.n
+      %tags               %.n
+      %tag-queries        %.n
+      %run-updates        %.y
+  ==
+::
+++  resource-for-update
+  |=  =vase
+  ^-  (unit resource:res)
+  =/  =update:store  !<(update:store vase)
+  ?-  -.q.update
+      %add-graph          `resource.q.update
+      %remove-graph       `resource.q.update
+      %add-nodes          `resource.q.update
+      %remove-nodes       `resource.q.update
+      %add-signatures     `resource.uid.q.update
+      %remove-signatures  `resource.uid.q.update
+      %archive-graph      `resource.q.update
+      %unarchive-graph    ~
+      %add-tag            ~
+      %remove-tag         ~
+      %keys               ~
+      %tags               ~
+      %tag-queries        ~
+      %run-updates        `resource.q.update
+  ==
+::
+++  initial-watch
+  |=  [=path =resource:res]
+  ^-  vase
+  ?>  (can-join:grp resource src.bowl)
```

Comment:
realistically, we should check membership here, not if they can join.

Reply:
Also needs to look up the group in the md-store.

Reply:
I've updated it. Does this mechanism of looking up the group in the `metadata-store` look alright?